searchlib/src/vespa/searchlib/tensor/hnsw_index.h

```diff
@@ -145,8 +145,7 @@ class HnswIndex : public NearestNeighborIndex {
 
     // Should only be used by unit tests.
     HnswNode get_node(uint32_t docid) const;
-
-    // TODO: Implement set_node() as well for use in unit tests.
+    void set_node(uint32_t docid, HnswNode node);
```

Comment:
Even though this is just used for unit testing, consider passing HnswNode as const reference.